service/forward_service.cc

```diff
@@ -519,7 +519,7 @@ future<query::forward_result> forward_service::dispatch(query::forward_request r
                     std::pair<netw::messaging_service::msg_addr, dht::partition_range_vector> vnodes_with_addr
                 ) {
                     netw::messaging_service::msg_addr addr = vnodes_with_addr.first;
-                    std::optional<query::forward_result>& result_ = result;
+                    query::forward_result& result_ = result;
```

Comment:
why do we need `result_`? What's the problem accessing `result` itself?
It's quite confusing.

Reply:
Avoids needing to patch users later.